src/shogun/util/converters.h

```diff
@@ -12,6 +12,15 @@
 
 #include <shogun/lib/common.h>
 
+#ifdef _MSC_VER
+#include <cfloat>
+// TODO: microsoft should really start supporting c++11
+#define IS_FINITE(x) _isfinite(x)
+#else
+#include <cmath>
```

Comment:
@vigsterkr this should solve your ruby issue

Reply:
otherwise you might have to include header at the top of the swig file, like [here](https://github.com/robotology/idyntree/blob/devel/bindings/iDynTree.i#L29)